lib/Component.lua

```diff
@@ -10,6 +10,14 @@ local Component = {}
 
 Component.__index = Component
 
+-- The error message that is thrown when setState is called in the wrong place.
+-- This is declared here to avoid really messy indentation.
+local INVALID_SETSTATE_MESSAGE = [[setState cannot be used currently, are you calling setState from any of:
+* the willUpdate or willUnmount lifecycle hooks
+* the init function
+* the render function
+* the shouldUpdate function]]
```

Comment:
Lua ignores the first newline after `[[` I'm fairly sure -- we should put one there to make the message more coherent!

Reply:
Oops, this should be fixed now.